File: py/verify.py
from algosdk.future import *
from algosdk.account import generate_account
from algosdk import encoding
from nacl.signing import SigningKey, VerifyKey
from nacl.exceptions import BadSignatureError
import base64


def verify_signed_transaction(stxn: transaction.SignedTransaction):

    if stxn.signature is None or len(stxn.signature) == 0:
        return False

    public_key = stxn.transaction.sender
    if stxn.authorizing_address is not None:
        public_key = stxn.authorizing_address

    verify_key = VerifyKey(encoding.decode_address(public_key))

    prefixed_message = b"TX" + base64.b64decode(
        encoding.msgpack_encode(stxn.transaction)
    )

    try:
        verify_key.verify(prefixed_message, base64.b64decode(stxn.signature))
        return True
    except BadSignatureError:
        return False


def main():
    sk, addr = generate_account()
    sp = transaction.SuggestedParams(
        0, 1, 2, "nDtgfcrOMvfAaxYZSL+gCqA2ot5uBknFJuWE4pVFloo="
    )
    txn = transaction.PaymentTxn(addr, sp, addr, 0)
    stxn = txn.sign(sk)
    print("Valid? ", verify_signed_transaction(stxn))


if __name__ == "__main__":
    main()

File: js/verify.ts
import algosdk, { makePaymentTxn, makePaymentTxnWithSuggestedParamsFromObject, SignedTransaction, SuggestedParams } from 'algosdk'
import { getAccounts } from './sandbox'
import nacl from 'tweetnacl'

function verifySignedTransaction(stxn: SignedTransaction){

    if (stxn.sig === undefined) return false;

    const pk_bytes = stxn.txn.from.publicKey

    const sig_bytes = new Uint8Array(stxn.sig);

    const txn_bytes = algosdk.encodeObj(stxn.txn.get_obj_for_encoding())
    const msg_bytes = new Uint8Array(txn_bytes.length + 2)
    msg_bytes.set(Buffer.from("TX"))
    msg_bytes.set(txn_bytes, 2)

    return nacl.sign.detached.verify(msg_bytes, sig_bytes, pk_bytes);

}

(function(){
    const sp: SuggestedParams = {fee: 0, firstRound: 1, lastRound:2, genesisHash:"nDtgfcrOMvfAaxYZSL+gCqA2ot5uBknFJuWE4pVFloo=", genesisID:"sandnet-v1"} 
    const acct = algosdk.generateAccount()
    const txn = makePaymentTxnWithSuggestedParamsFromObject({
        from: acct.addr,
        to: acct.addr,
        amount: 0,
        suggestedParams: sp,
    })

    const blob = txn.signTxn(acct.sk)

    const stxn = algosdk.decodeSignedTransaction(blob)
    console.log("Valid? ", verifySignedTransaction(stxn))
})()